### PR TITLE
Add libpas

### LINKED ISSUE #130: Add libpas comparison
A new malloc() has appeared:  https://github.com/WebKit/WebKit/blob/main/Source/bmalloc/libpas/Documentation.md
Some discussion:  https://news.ycombinator.com/item?id=31581727
There's some interesting things going on with the `always_inline` parts.

### PR DESCRIPTION
**Pull request by jvoisin**
This should close #130

cc @pizlonator

### CODE REVIEW

File: bench.sh [hunk 1]
```diff
@@ -108,6 +108,7 @@ alloc_lib_add "hm"     "$localdevdir/hm/out/libhardened_malloc$extso"
 alloc_lib_add "hml"    "$localdevdir/hm/out-light/libhardened_malloc-light$extso"
 alloc_lib_add "iso"    "$localdevdir/iso/build/libisoalloc$extso"
 alloc_lib_add "je"     "$localdevdir/jemalloc/lib/libjemalloc$extso"
+alloc_lib_add "lp"     "$localdevdir/lp/Source/bmalloc/libpas/build-cmake-default/Release/libpas_lib.$extso"
```
Comment (mjp41) [inline bench.sh:111]:
```suggestion
alloc_lib_add "lp"     "$localdevdir/lp/Source/bmalloc/libpas/build-cmake-default/Release/libpas_lib$extso"
```

**Review by mjp41 — APPROVED** [review]
LGTM, just skimming through the results in the GitHub action it doesn't look particularly performance competitive on this benchmark suite.

File: .github/workflows/all.yml [hunk 2]
```diff
@@ -80,7 +80,8 @@ jobs:
       - name: Install and build all benchmarks and allocators
         # ff: error: use of undeclared identifier 'MAP_POPULATE'
         # gd: so many errors
-        run: ./build-bench-env.sh all no-lean no-gd no-ff
+        # lp: TODO(jvoisin)
```
Comment (mjp41) [inline .github/workflows/all.yml:83]: Is Mac still broken? Skimming the actions I couldn't see anything Mac specific that has failed?

**Review by mjp41 — APPROVED** [review]
LGTM

File: .github/workflows/all.yml (hunk 2 above)
Comment (jvoisin) [reply .github/workflows/all.yml:83]: It's not broken, just disabled for now, since it was getting late here :)
I'll add osx support in a followup PR.

**Comment by jvoisin** [comment]
About the performances, the author ( @pizlonator ) said:

> Libpas beats other mallocs in WebKit. I also had benchmarks involving non-WebKit workloads and the results were all over the place (sometimes slower than other mallocs by a lot, sometimes faster by a lot - same thing with memory, sometimes more efficient, sometimes less). This didn't surprise me; I've seen this before when writing memory management code.

**Comment by mjp41** [comment]
Yeah, I was wondering how possible it would be to add this workload to mimalloc-bench.  But the tests don't look entirely self-contained.

**Pull request merged**